src/components/fields/contact-number/contact-number.tsx

```diff
@@ -0,0 +1,170 @@
+
+						return PhoneHelper.isInternationalNumber(selectedCountry?.name, value);
+					}
+				),
+			validation
+		);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [validation, selectedCountry]);
+
+	useEffect(() => {
+		setStateValue(value || "");
+	}, [value]);
+
+	useEffect(() => {
+		const countryName = country || "Singapore";
+
+		setSelectedCountry({
+			name: countryName,
+			prefix: getPrefix(countryName),
+		});
+	}, [country]);
+
+	// =============================================================================
+	// EVENT HANDLERS
+	// =============================================================================
+	const handlePrefixChange = (key: TCountry, prefix: string): void => {
+		const { number: phoneNumberWithoutPrefix } = PhoneHelper.getParsedPhoneNumber(stateValue);
+		const phoneNumberWithPrefix = PhoneHelper.formatPhoneNumber(prefix, phoneNumberWithoutPrefix);
+
+		setSelectedCountry({ name: key, prefix });
+		onChange({ target: { value: phoneNumberWithPrefix } });
+	};
+
+	const handleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+		const phoneNumberWithPrefix = PhoneHelper.formatPhoneNumber(selectedCountry?.prefix, event.target.value);
+
+		onChange({ target: { value: phoneNumberWithPrefix } });
+	};
+
+	// =============================================================================
+	// HELPER FUNCTIONS
+	// =============================================================================
+	const formatDisplayValue = (): string => {
+		const parsedValues = stateValue.split(" ");
+		return parsedValues.length > 1 ? parsedValues[1] : stateValue;
+	};
+
+	const getSpacing = (): number => {
+		if (selectedCountry?.name === "Singapore") return 4;
+
+		return 0;
+	};
+
+	const getMaxLength = (): number => {
+		if (selectedCountry?.name === "Singapore") {
+			return 9;
+		}
+		return 14;
+	};
+
+	const getAddOns = (): AddonProps<string, unknown> => {
+		if (!allowInternationalNumbers) {
+			return { attributes: { value: "+65" } };
+		}
+		return {
+			type: "list",
+			attributes: {
+				options: getCountries(),
+				listExtractor: (item: string) => `${item} (${getPrefix(item)})`,
+				enableSearch,
+				onSelectOption: handlePrefixChange,
+				value: selectedCountry
+					? getCountryFromPrefix(selectedCountry.prefix)
+					: InternationalCallingCodeMap["Singapore"],
+				valueExtractor: (item: string) => getPrefix(item),
+			},
+		};
+	};
+
+	// =============================================================================
+	// RENDER FUNCTIONS
+	// =============================================================================
+	return (
+		<Form.InputGroup
+			{...otherSchema}
+			{...otherProps}
+			id={id}
+			name={name}
+			disabled={disabled}
+			type="tel"
+			inputMode="tel"
+			label={label}
+			value={formatDisplayValue()}
+			onChange={handleChange}
+			addon={getAddOns()}
+			errorMessage={error?.message}
+			placeholder={placeholder || "Enter mobile number"}
```

Comment:
need to infer whether home or mobile phone is expected through the `validation` config